src/App.tsx

```diff
@@ -1,223 +1,286 @@
+  const [clearLoader, setClearLoader] = useState(false);
+  const [loadingAllTodos, setLoadingAllTodos] = useState(false);
+  const [loadTodo, setLoadTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<ErrorType>(ErrorType.noError);
+
+  const handleHidden = useCallback(
+    (value: boolean) => {
+      setHiddenErrorTable(value);
+    }, [],
+  );
+
+  const setComplitedTodo = async (todo: Todo) => {
+    await updateTodo(todo, { completed: true });
+  };
+
+  const setUnComplitedTodo = async (todo: Todo) => {
+    await updateTodo(todo, { completed: false });
+  };
+
+  const handleLoadTodo = useCallback((value: Todo | null) => {
+    setLoadTodo(value);
+  }, []);
+
+  const handleTypeFilter = useCallback((type: string) => {
+    setTypeFilter(type);
+  }, []);
+
+  const hanldeUpdateError = useCallback((value: ErrorType) => {
+    setError(value);
+  }, []);
+
+  const handlerRemoveError = useCallback((value: ErrorType) => {
+    setError(value);
+  }, []);
+
+  const handlerNewTitle = useCallback((value: string) => {
+    setNewTodoTitle(value);
+  }, []);
+
+  const getUpdateTodoList = useCallback((uSer:User) => {
+    getTodos(uSer.id).then(response => {
+      setTodoList(response);
+      setIsAdding(null);
+      setClearLoader(false);
+      setLoadTodo(null);
+      setLoadingAllTodos(false);
+    });
+  }, [user]);
+
+  const createTodo = useCallback(
+    async (event:React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (!newTodoTitle.trim()) {
+        setError(ErrorType.errorEmptyTitile);
+        setHiddenErrorTable(false);
+
+        return;
+      }
+
+      if (user) {
+        const newTodo = {
+          userId: user.id,
+          title: newTodoTitle,
+          completed: false,
+        };
+
+        setIsAdding(newTodo);
+
+        try {
+          await createTodos(newTodo);
+        } catch {
+          setError(ErrorType.errorAdd);
+          setHiddenErrorTable(false);
+        }
+
+        getUpdateTodoList(user);
+        setNewTodoTitle('');
+      }
+    }, [newTodoTitle],
+  );
+
+  const checkedAllCompletedTodo = (list: Todo[]) => (
+    list.every(todo => todo.completed)
+  );
+
+  const selectComplited = useCallback(async (todo:Todo) => {
+    try {
+      setLoadTodo(todo);
+      if (todo.completed) {
+        await setUnComplitedTodo(todo);
+      } else {
+        await setComplitedTodo(todo);
+      }
+    } catch {
+      setError(ErrorType.errorUpdate);
+      setHiddenErrorTable(false);
+      setLoadTodo(null);
+    }
+
+    if (user) {
+      getUpdateTodoList(user);
+    }
+  }, [todoList]);
+
+  const selectAllTodos = useCallback(() => {
+    setLoadingAllTodos(true);
+    todoList.map(async (todo) => {
+      if (!checkedAllCompletedTodo(todoList)) {
+        if (!todo.completed) {
+          selectComplited(todo);
+        }
+      } else {
+        selectComplited(todo);
+      }
+    });
```

Comment:
use forEach method instead of the map one, if you don't need the result of map method